server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistComponentsStep.java

```diff
@@ -335,46 +385,25 @@ private static void setParentModuleProperties(ComponentDto componentDto, PathAwa
 
   }
 
-  private static boolean updateExisting(ComponentDto existingComponent, ComponentDto newComponent) {
-    boolean modified = false;
-    if (!StringUtils.equals(existingComponent.name(), newComponent.name())) {
-      existingComponent.setName(newComponent.name());
-      modified = true;
-    }
-    if (!StringUtils.equals(existingComponent.longName(), newComponent.longName())) {
-      existingComponent.setLongName(newComponent.longName());
-      modified = true;
-    }
-    if (!StringUtils.equals(existingComponent.description(), newComponent.description())) {
-      existingComponent.setDescription(newComponent.description());
-      modified = true;
-    }
-    if (!StringUtils.equals(existingComponent.path(), newComponent.path())) {
-      existingComponent.setPath(newComponent.path());
-      modified = true;
-    }
-    if (!StringUtils.equals(existingComponent.moduleUuid(), newComponent.moduleUuid())) {
-      existingComponent.setModuleUuid(newComponent.moduleUuid());
-      modified = true;
-    }
-    if (!existingComponent.moduleUuidPath().equals(newComponent.moduleUuidPath())) {
-      existingComponent.setModuleUuidPath(newComponent.moduleUuidPath());
-      modified = true;
-    }
-    if (!ObjectUtils.equals(existingComponent.getRootUuid(), newComponent.getRootUuid())) {
-      existingComponent.setRootUuid(newComponent.getRootUuid());
-      modified = true;
-    }
-    if (!ObjectUtils.equals(existingComponent.getCopyResourceUuid(), newComponent.getCopyResourceUuid())) {
-      existingComponent.setCopyComponentUuid(newComponent.getCopyResourceUuid());
-      modified = true;
-    }
-    if (!existingComponent.isEnabled()) {
-      // If component was previously removed, re-enable it
-      existingComponent.setEnabled(true);
-      modified = true;
+  private static Optional<ComponentUpdateDto> compareForUpdate(ComponentDto existing, ComponentDto target) {
+    boolean hasDifferences = !StringUtils.equals(existing.getCopyResourceUuid(), target.getCopyResourceUuid()) ||
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (9) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1067)